graphics/tinygl/zgl.h

```diff
@@ -452,6 +452,10 @@ struct GLContext {
 
 	bool _enableDirtyRectangles;
 
+	// stipple
+	bool polygon_stipple_enabled;
+	void *polygon_stipple_pattern;
```

Comment:
I think this is wrong, it looks like this is storing a pointer to the stipple pattern instead of a copy of it, which means the stipple pattern memory has to remain valid as long as the stipple pattern is used.  I don't think this matches the semantics of `glPolygonStipple`.

Reply:
stiple pattern can be simple array on upper layer

Reply:
I made some changes but I'm not sure where exactly we could have the stipple array, if not in the GL context.